File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/FieldMessageIdLiveDataValidationListener.java
package com.denchic45.kts.uivalidator.validatorlistener;

import androidx.lifecycle.MutableLiveData;

import com.denchic45.kts.Pair;
import com.denchic45.kts.uivalidator.Rule;

public class FieldMessageIdLiveDataValidationListener extends LiveDataValidationListener<Pair<Integer, Integer>> {

    private final int id;

    public FieldMessageIdLiveDataValidationListener(int id, MutableLiveData<Pair<Integer, Integer>> mutableLiveData) {
        super(mutableLiveData);
        this.id = id;
    }

    @Override
    void onSuccess() {
        mutableLiveData.setValue(new Pair<>(id, null));
    }

    @Override
    void onError(Rule rule) {
        mutableLiveData.setValue(new Pair<>(id, rule.getErrorResId()));
    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/validatorlistener/FieldMessageIdLiveDataValidationListener.kt
package com.denchic45.kts.uivalidator.validatorlistener

import com.denchic45.kts.SingleLiveData
import com.denchic45.kts.uivalidator.Rule

class FieldMessageIdLiveDataValidationListener(
    private val id: Int,
    mutableLiveData: SingleLiveData<Pair<Int, Int?>>
) : LiveDataValidationListener<Pair<Int, Int?>>(mutableLiveData) {
    override fun onSuccess() {
        mutableLiveData.value = Pair<Int, Int?>(id, null)
    }

    override fun onError(rule: Rule) {
        mutableLiveData.value = Pair(id, rule.errorResId)
    }
}